airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeGcsStreamCopier.java

```diff
@@ -30,20 +41,49 @@ public SnowflakeGcsStreamCopier(final String stagingFolder,
     this.filenameGenerator = stagingFilenameGenerator;
   }
 
+  @Override
+  public void copyStagingFileToTemporaryTable() throws Exception {
+    List<List<String>> partition = Lists.partition(new ArrayList<>(gcsStagingFiles), MAX_CHUNK_SIZE);
+    for (int i = 0; i < partition.size(); i++) {
+      LOGGER.info("Starting copy chunk {} to tmp table: {} in destination for stream: {}, schema: {}. Chunks count {}", i, tmpTableName, streamName,
+          schemaName, partition.size());
+      executeCopy(partition.get(i));
```

Comment:
I think this will still execute in serial - we want each of these `executeCopy` calls to happen in parallel.

Reply:
> I think this will still execute in serial - we want each of these `executeCopy` calls to happen in parallel.

do you mean something like that?
`      executorService.execute(()->executeCopy(partition.get(i)));
`

Reply:
yep!

Reply:
also fyi that we're limiting snowflake to 5 open connections (https://github.com/airbytehq/airbyte/pull/10342) so this should be a 5-thread pool. cc @tuliren

Reply:
The max number of connections is defined here: https://github.com/airbytehq/airbyte/blob/master/airbyte-db/lib/src/main/java/io/airbyte/db/Databases.java#L211

Feel free to bump this number if needed. We should choose the number of threads based on needs and memory limitation.

Reply:
@edgao added multithreading